### ISSUE TITLE
`best.h5` is not the actual best model after resuming training

### ISSUE THREAD

**Issue by 3tz** [3tz/ganabi#3]
After resuming training, the training session resets the default best validation loss to `inf` which means the model after the 1st epoch from resuming training is saved and the models that perform better than this model will be used to replace it thereafter. However, `ckpts` still have the correct models and `training.log` files still contain the correct accuracy and loss for all of the epochs, and the actual best `.h5` file can be found with them. 

Credit to Duong Nguyen for reporting a suspicious behavior during the training process that eventually leads to the discovery of this bug.
